rotkehlchen/db/dbhandler.py

```diff
@@ -1907,7 +1929,14 @@ def get_trades(
 
         trades = []
         for result in results:
+            location = deserialize_location_from_db(result[2])
             try:
+                # NB: Required for AMM trades (e.g. Uniswap)
+                fee_currency = (
```

Comment:
@LefterisJP as far as I'm aware, this is the less invasive I can be with the current trades logic, and reusing as much as possible existing trade DB CRUD methods (i.e. not implementing new "add/get trades with custom identifier for uniswap location").
If you check my previous commit you will see the whole change of strategy for interacting with the DB.

The fact that Trade `fee_currency` is an `<Asset>`, and IDs are not stored in the instance but generated on the fly (via property) has made the whole thing quite tricky to manage.

Reply:
as i said in another part i think this whole approach is quite hacky. I really don't see why you can't have a different table for the uniswap trades. It will make the logic much easier to handle no?